File: 2582-pass-the-pillow/2582-pass-the-pillow.py
class Solution(object):
    def passThePillow(self, n, time):
        """
        :type n: int
        :type time: int
        :rtype: int
        """
        if n>time:
            return time+1
        remaining=time%(n-1)
        if ((time//(n-1))%2)==0:
            return remaining+1
        return n-remaining
        

File: 2582-pass-the-pillow/2582-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
        if(n>time)
            return time+1;
        int remain=time%(n-1);
        if((time/(n-1)%2)==0)
            return remain+1;
        return n-remain;
    }
}